src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Utility/FileWatchers/SolutionConfigFileWatcher.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.IO;
+using NuGet.Configuration;
+
+namespace NuGet.PackageManagement.VisualStudio.Utility.FileWatchers
+{
+    internal sealed class SolutionConfigFileWatcher : IFileWatcher
+    {
+        private bool _disposed;
+        private FileSystemWatcher _solutionWatcher;
+        private FileSystemWatcher? _dotNuGetWatcher;
+
+        public SolutionConfigFileWatcher(string solutionDirectory)
+        {
+            string? current = solutionDirectory.EndsWith("\\", StringComparison.Ordinal)
+                ? Path.GetDirectoryName(solutionDirectory)
+                : solutionDirectory;
+
+            _solutionWatcher = Create(solutionDirectory);
+
+            var dotNuGetPath = Path.Combine(solutionDirectory, NuGetConstants.NuGetSolutionSettingsFolder);
+            if (Directory.Exists(dotNuGetPath))
+            {
+                _dotNuGetWatcher = Create(dotNuGetPath);
+            }
+
+            FileSystemWatcher Create(string path)
+            {
+                var watcher = new FileSystemWatcher(path, Settings.DefaultSettingsFileName);
```

Comment:
I can see we have [NuGetDefaults.Config](https://learn.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior#nuget-defaults-file) file too. Do we need to watch it too?

Reply:
`NuGetDefaults.Config` is not in a user-writable location, it needs admin access to modify. So, there are no actions in Visual Studio that will modify the file (branch change, or using the package source or package source mapping options windows). Similarly, `dotnet nuget` and `nuget.exe` commands won't edit the file unless the customer both elevates to admin and also uses `--config-file` to explicitly tell nuget to modify that specific file.

So, the scenarios where the file will be modified are very unlikely (frequency of nearly zero), and I believe customers are more accepting that programs need to be restarted when modifying machine wide state.

Although having written all this, I assume that `NuGetDefaults.config` is the preferred way for admins to force certain compliance via group policy. Therefore, in theory a group policy update could update the file while VS is running. On the other hand, NuGet allows users to override any NuGetDefaults.config with their own user-writable nuget.config, so the group policy compliance argument falls flat, in my opinion.

What do others think?